python/lib/communication/dmod/communication/maas_request/ngen/abstract_nextgen_request.py

```diff
@@ -311,7 +311,7 @@ def output_formats(self) -> List[DataFormat]:
         List[DataFormat]
             List of the formats of each required output dataset for the requested job.
         """
-        return [DataFormat.NGEN_OUTPUT]
+        return [DataFormat.ARCHIVED_NGEN_CSV_OUTPUT]
```

Comment:
An interesting question.  Probably something that deserves its own issue, as it isn't well defined at the moment.

Reply:
This may be simpler than I thought initially (although we can still consider giving it its own issue if more conversation is needed).

First, it is not possible to completely, explicitly define an ngen output data format that covers all possible output variables.  I'm not going to work through it entirely right now, but my back-of-the-envelop math says ngen may actually support uncountably infinitely many possible combinations of outputs, in theory at least.  Regardless, too many to cover.  

And so, we don't try to.  While some _DataFormat_ values do define `data_fields`, the ngen output formats do not.  The implication is that we cannot necessarily completely summarize what could be in there, so users of the data need to determine that for themselves.

A format does have to enable domains to elaborate on certain things, though, via the standard indices.  So it is useful (perhaps necessary) to distinguish particulars of how the indices data are available within the dataset, and this is done implicitly by the format and its name.  E.g., for `ARCHIVED_NGEN_CSV_OUTPUT`, there will be a bunch of CSV inside an archive, with catchment ids in the names of the CSVs and times (and other data) inside the CSVs. 

So, it is still useful to have ngen-related output formats that imply only part of how data are structured within such datasets, even if it doesn't (because it can't) tell us everything about what's in there.